.devcontainer/Dockerfile

```diff
@@ -44,6 +54,19 @@ RUN wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add - 2>/dev/
 # Set the default clang-tidy, so CMake can find it
 RUN update-alternatives --install /usr/bin/clang-tidy clang-tidy $(which clang-tidy-${LLVM_VER}) 1
 
+# Set clang-${LLVM_VER} as default clang
+RUN update-alternatives --install /usr/bin/clang clang $(which clang-${LLVM_VER}) 100
+RUN update-alternatives --install /usr/bin/clang++ clang++ $(which clang++-${LLVM_VER}) 100
+
+# Allow the user to set compiler defaults
+ARG USE_CLANG
+# if --build-arg USE_CLANG=1, set CC to 'clang' or set to null otherwise.
+ENV CC=${USE_CLANG:+"clang"}
+ENV CXX=${USE_CLANG:+"clang++"}
+# if CC is null, set it to 'gcc' (or leave as is otherwise).
+ENV CC=${CC:-"gcc"}
+ENV CXX=${CXX:-"g++"}
```

Comment:
When I wrote this, I purposefully placed this argument at the very end of the file so that users can turn `USE_CLANG` on or off without needing to rebuild much of anything at all. Now that it has been moved up, if you change `USE_CLANG` and rebuild the image, the cache for `cmake` and all of the optional dependencies is invalidated, and will be installed again.

I prefer this section at the end of the file.

Reply:
Hmm yes it is better at the end